cql3/statements/select_statement.cc

```diff
@@ -461,7 +461,7 @@ generate_base_key_from_index_pk(const partition_key& index_pk, const std::option
         if (!view_col) {
             throw std::runtime_error(format("Base key column not found in the view: {}", base_col.name_as_text()));
         }
-        if (base_col.type != view_col->type) {
+        if (base_col.type->without_reversed() != *view_col->type) {
```

Comment:
Is it possible for view_col->type to be reversed too?

Reply:
btw, I think it's a mistake to keep the reversed attribute in the type. We move it outside, to some attributes around the clustering key. It's really a marker, not a change to the type itself. Of course this is a lot of work for small payoff, I'm not suggesting we do it.

Reply:
Yesterday I thought it can't be reversed, because during index creation the index column becomes the partition key, but the matter might be more complicated for local indexes. There, we keep the original partition key and the indexed column becomes the first clustering key. For now we don't care much, because we always ask for a specific value of the indexed column, so it doesn't matter if it's reversed or not. However, if we ever decide to implement https://github.com/scylladb/scylla/issues/5547 , it will start to matter. Then we'd also have to introduce a way of determining what kind of clustering order the user is interested in. `CREATE INDEX ON t((p),v desc)`?

But, anyway, it's not particularly costly to also call `without_reversed` on the view_col type to be double sure, so I'll send a follow-up patch. Even if it's not reproducible now, it will make the code more future-proof.